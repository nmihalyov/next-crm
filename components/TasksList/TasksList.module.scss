.empty {
  opacity: 0.5;
  text-align: center;
}