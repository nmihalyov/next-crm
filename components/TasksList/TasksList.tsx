import { useCallback } from 'react';
import type { Task } from '../../types/task';
import { Empty } from 'antd';

import TaskCard from '../_ui/TaskCard/TaskCard';

const TasksList: React.FC<{
  tasks: Array<Task>,
  onPatch(id: number, completed: boolean): void,
  onRemove(id: number): void
}>= props => {
  const { tasks, onPatch, onRemove } = props;

  const memoizedOnPatch = useCallback(onPatch, []);
  const memoizedOnRemove = useCallback(onRemove, []);

  return (
    <div>
      {tasks.length ? tasks.map(task =>
        <TaskCard
          key={task.id}
          onPatch={memoizedOnPatch}
          onRemove={memoizedOnRemove}
          {...task} />
      ) : <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} description="No tasks yet" />}
    </div>
  );
};

export default TasksList;