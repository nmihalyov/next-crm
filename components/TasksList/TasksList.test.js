import TasksList from './TasksList';

describe('TasksList:', () => {
  const setUp = props => shallow(<TasksList {...props} />);
  const tasks = [{userId: 1, id: 1, title: 'Task title', completed: true}];

  it('renders component if any tasks', () => {
    const component = setUp(tasks);

    expect(component).toMatchSnapshot();
  });

  it('renders empty element if no tasks', () => {
    const component = setUp();
    const empty = component.find('Empty');

    expect(empty).toHaveLength(1);
  });
});