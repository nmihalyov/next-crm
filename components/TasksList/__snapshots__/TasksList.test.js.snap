// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksList: renders component if any tasks 1`] = `
<div>
  <Memo(TaskCard)
    completed={true}
    id={1}
    key="1"
    title="Task title"
    userId={1}
  />
</div>
`;

exports[`TasksList: renders empty element if no tasks 1`] = `
<Empty
  description="No tasks yet"
  image={<Simple />}
/>
`;
